cargo-dist/src/lib.rs

```diff
@@ -214,6 +304,17 @@ fn generate_source_tarball(
     Ok(())
 }
 
+fn generate_fake_source_tarball(
+    _graph: &DistGraph,
+    _committish: &str,
+    _prefix: &str,
+    target: &Utf8Path,
+) -> DistResult<()> {
+    std::fs::File::create(target)?;
```

Comment:
is there any reason this should be faked? i thought source tarballs were trivial to generate on any platform.

Reply:
Source tarballs require git to be installed and for you to be running the process inside a git checkout. Faking this makes it possible for us to simulate it in a few extra circumstances.

Reply:
this is kinda overkill but i'll defer to your judgement :)

Reply:
For context, this fixes a test for nix - I'd planned to write a fix for this in other contexts, and it turned out this let me do it as part of this PR.